iterator: compilation error when defining a custom iterator

blocks https://github.com/lightpanda-io/browsercore/pull/74 

I looks like I can't return custom pointer but I'm not sure...

```
vendor/jsruntime-lib/src/engines/v8/types_primitives.zig:14:5: error: parameter of opaque type '.home.pierre.wrk.browsercore.zig-cache.o.732a86b50a20b2fa4e881336d2ab7e9a.cimport.struct_dom_html_element' not allowed
/home/pierre/wrk/browsercore/zig-cache/o/732a86b50a20b2fa4e881336d2ab7e9a/cimport.zig:1511:37: note: opaque declared here
vendor/jsruntime-lib/src/engines/v8/generate.zig:383:9: error: expected type 'netsurf.Unknown', found '*netsurf.Unknown'
src/netsurf.zig:646:21: note: struct declared here
```

--- a/src/reflect.zig
+++ b/src/reflect.zig
@@ -667,6 +667,8 @@ pub const StructNested = struct {
             return false;
         } else if (std.mem.eql(u8, decl.name, "mem_guarantied")) {
             return false;
+        } else if (std.mem.eql(u8, decl.name, "Exception")) {
+            return false;
         }
 
         // exclude declarations who are not types
@@ -776,6 +778,11 @@ pub const Struct = struct {
         }
         inline for (self.methods) |*method| {
             try method.lookupTypes(structs);
+        }
+        inline for (self.nested) |nested| {
+            inline for (nested.fields) |*field| {
+                try field.lookup(structs);
+            }
         }
     }
 

